### PR TITLE
slack_menu empty tag list fix; uvloop support.

### PR DESCRIPTION
**Pull request by 3c-un-str**
Add uvloop support to run the async event loop; uvicorn will pick this over asyncio, if available. Not sure about others, but In my setup (ubuntu VM), not having uvloop was resulting in 8/10 calls failing (some like /dispatch-update-incident failed 10/10) to Slack's API with the 'expired_trigger_id' error. I still see that error (w/uvloop), but it is more like 1/10. Also, this is an configurable option: "--loop [auto|asyncio|uvloop]    Event loop implementation.  [default: auto]"

Also, including the commit from earlier today which fixes slack_menu returning empty tag list from non-incident channels. Not sure how that was missing in the master branch although. 

Some minor str changes.

### CODE REVIEW

**Comment by mvilanova** [comment]
Thanks for the PR. I'll take a look at it as soon as I can.

File: src/dispatch/evergreen/scheduled.py [hunk 1]
```diff
@@ -54,15 +54,24 @@ def create_evergreen_reminder(
...
-        owner_email,
-        notification_text,
-        notification_template,
-        notification_type,
-        name=name,
-        subject=subject,
-        items=items,  # plugin expect dicts
-    )
+
+    # Gmail API seems to constantly throw "tenacity.RetryError: RetryError", even though the email(s)
```
Comment (mvilanova) [inline src/dispatch/evergreen/scheduled.py:58]: Interesting... We are not experiencing the same behavior, but I'll take a look at our server logs to double check. Any useful errors in your GCP project that could shed light at what is happening?

**Comment by mvilanova** [comment]
Can we split this PR into two: one for the Gmail protections and another one for the Slack stuff?

**Comment by mvilanova** [comment]
The Slack and str changes look good. I can merge them once the Gmail ones are moved to a separate PR.

**Comment by 3c-un-str** [comment]
dropped the gmail changes from this PR. Will submit a new one.

**Pull request merged**